x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,317 @@
+Some validation is customizable via <<ref-jwt-settings>>.
+
+JWS header `alg` is required, and must be in the realm's allow list. JWS header `typ` is optional.
+
+JWT claims `iss`, `aud`, `sub`, `exp`, `iat`, `nbf`, and `auth_time` are validated using OpenID Connect ID Token validation rules.
+JWT Claims `principal`, `groups`, `name`, `mail`, and `dn` are validated using {es} validation rules.
+
+IMPORTANT: `nonce` claim is ignored; it is only applicable inside an OIDC workflow, not application to JWT realm.
+
+NOTE: Validation of the JWT time-based claims `exp`, `iat`, `nbf`, and `auth_time` can be relaxed via the realm setting `allowed_clock_skew`.
+
+* The `iss` claim from the JWT is required. It must be an exact, case-sensitive match to the realm setting `allowed_issuer`.
+
+* The `aud` claim from the JWT is required. It is a comma separated value (CSV). One of the values must be an exact,
+case-sensitive match to any of the CSV values in the realm setting `allowed_audiences`.
+
+* The `exp` claim from the JWT is required. It is UTC milliseconds since epoch, and must be after wall clock.
+
+* The `iat` claim from the JWT is required. It is UTC milliseconds since epoch, and must be before wall clock.
+
+* The `nbf` claim from the JWT is optional. If present, it is UTC milliseconds since epoch, and must be before wall clock.
+
+* The `auth_time` claim from the JWT is optional. If present, it is UTC milliseconds since epoch, and must be before wall clock.
+
+* The _principal_ claim from the JWT is required, and defaults to `sub`. It is configurable via the realm setting `claims.principal`.
+An optional regular expression can be configured via the realm setting `claims.principal_pattern` to extract a substring.
+
+* The _groups_ claim from the JWT is optional. It is configurable via the realm setting `claims.groups`.
+An optional regular expression can be configured via the realm setting `claims.groups_pattern` to extract a substring value.
+
+* The _name_ claim from the JWT is optional. It is configurable via the realm setting `claims.name`.
+An optional regular expression can be configured via the realm setting `claims.name_pattern` to extract a substring value.
+
+* The _mail_ claim from the JWT is optional. It is configurable via the realm setting `claims.mail`.
+An optional regular expression can be configured via the realm setting `claims.mail_pattern` to extract a substring value.
+
+* The _dn_ claim from the JWT is optional. It is configurable via the realm setting `claims.dn`.
+An optional regular expression can be configured via the realm setting `claims.dn_pattern` to extract a substring value.
+
+[[jwt-authorization]]
+==== JWT Realm Authorization
+
+JWT realm authorization supports role mapping API or delegated authorization realms.
+
+WARNING: JWT realm does not support using the role_mapping.yml file.
+
+==== JWT Realm Authorization via Role Mapping API
```

Comment:
```suggestion
===== JWT realm authorization via Role Mapping API
```

This should be nested under the previous section

Reply:
I'm still working through these latter sections and will next them accordingly.